### ISSUE TITLE
The old searchprovider overwrites the new defaultProvider.

### ISSUE THREAD

**Issue by SrTobi** [SrTobi/code-bing#3]
Thats not what we want.

**Comment by jahands** [comment]
That part was designed like that so that it won't break for users who installed 0.0.1.

I can remove that part if you are fine with it breaking the old `searchprovider` config option.
This was purely for backwards compatibility.

**Comment by SrTobi** [comment]
No, what I mean is if both `searchprovider` and `defaultProvider` are defined, then `searchprovider` is chosen. I think that is not what we want. The problem is, that I can not think of a solution other then setting the default of `searchprovider` to `null`, which is not very nice.

**Comment by jahands** [comment]
Updated. the old `searchprovider` is supposed to be semi-deprecated, but still works for compatibility.

#5 will change it so that if the user changes defaultProvider to use something besides bing, then `searchprovider` will simply be ignored.

**Comment by SrTobi** [comment]
Works, but looks a little bit dirty. Instead of weired behavior, we could also let it like it is and show a warning + open the settings file:

```
    if (oldSearchProvider != null) {
        vscode.window.showWarningMessage("bla bla!", "Open global config").then(() => {
            vscode.commands.executeCommand("workbench.action.openGlobalSettings");
        });
    }
```

**Comment by jahands** [comment]
Good point. I'll add it to this PR and revert the workaround.

**Comment by jahands** [comment]
How's this?

```
    let oldSearchProvider = config.get<string>("searchprovider");
    if (oldSearchProvider != null) {
        defaultProvider = oldSearchProvider;
        vscode.window.showWarningMessage("codebing.searchprovider is depricated!", "Open global config").then(() => {
            vscode.commands.executeCommand("workbench.action.openGlobalSettings");
        });
    }
```

**Comment by jahands** [comment]
Fixed in #5

**Comment by SrTobi** [comment]
fixed in #5.
